src/robots/gazebo_uav.cc

```diff
@@ -0,0 +1,65 @@
+// BoB robotics includes
+#include "robots/gazebo_uav.h"
+#include "third_party/units.h"
+using namespace std::literals;
+using namespace units::literals;
+namespace BoBRobotics {
+namespace Robots {
+
+void GazeboQuadCopter::takeOff()
+{
+    m_Armed=true;
+    m_Thrust = 1;
+    sendCommand();
+    std::this_thread::sleep_for(500ms);
+    m_Thrust = 0.5;
+    sendCommand();
```

Comment:
I'm worried there might be some race conditions going on here. If someone presses the button for take-off and then presses some other buttons at the same time then things could be in an inconsistent state. You can sort this with a mutex.

Reply:
Put in a mutex on the takeoff and land functions. There's also an atomic_bool variable `is_Armed` that is checked in all the flight control functions

Reply:
Looks good to me.